lib/acts_as_ar_model.rb

```diff
@@ -142,8 +142,30 @@ def self.reflections
   # Find routines
   #
 
+  # This method is called by QueryRelation upon executing the query.
+  #   Since it will receive non-legacy search options, we need to convert
+  #   them to handle the legacy find.
+  def self.search(mode, options = {})
```

Comment:
> The original code just called standard active record api find.

Was thinking about this more and although the original code did this it was against the "old" find interface, which, moving forward, I don't want QueryRelation to abide by.  The "current" find interface for ActiveRecord only accepts ids.  As such, QueryRelation expects a different method, intentionally not `find` so the caller (or perhaps the QueryRelation::Queryable mixin) can implement an id-based find.  The method name I chose was `search` after some negotiation with @abellotti, with the purpose being to take the options from QueryRelation and use them to do the search against the source.

In this case of ActsAsArModel, it takes the options, converts them to legacy options and calls the legacy find.  Ideally, I'd like to kill the legacy find method of ActsAsArModel, and then kill the to_legacy_options method as well, moving towards the "current" ActiveRecord like interface.